### PR TITLE
Change order of slides and add sections to slides

### PR DESCRIPTION
**Pull request by kaylielau**
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)

There were some adjustments made to the order of topics and also certain content has been removed. However, some of the modifications have left the content a bit disjointed. For instance, in the slides titled "02_reality_control_flow_iterables," there are examples involving functions (which aren't introduced until slide 07_functions.ipynb) and string formatting (which has been removed altogether). As another example, Assignment #1 involved the use of string methods, but this content was completely removed from the slides as well. Inconsistencies like these cropped up in multiple places so I’m submitted this pull request to reorganize the slides and reinstate some of the missing content. 

## What did you learn from the changes you have made?
N/A

## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
N/A

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
N/A

## How were these changes tested?
N/A

## A reference to a related issue in your repository (if applicable)
N/A

## @mentions of the person or team responsible for reviewing proposed changes (At least 2 people)
@RohanAlexander @curtissfnorman @danielrazavi

## Checklist
- [ Y] I can confirm that my changes are working as intended

### CODE REVIEW

File: 01_slides/05_strings.ipynb [hunk 1]
```diff
@@ -609,6 +613,256 @@
...
+      "source": [
+        "len('Sometimes we type long-winded sentences to meet word counts or character counts.')"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "id": "659dd1a9",
+      "metadata": {},
+      "source": [
+        "## String methods\n",
```
Comment (danielrazavi) [inline 01_slides/05_strings.ipynb:671]: Thoughts on introducing this concept in the Object-Oriented programming lesson?

**Review by curtissfnorman — APPROVED** [review]
The sections I have reviewed (updates to the README) are all fine and ready to merge.

File: 01_slides/05_strings.ipynb [hunk 2]
```diff
@@ -609,6 +613,256 @@
...
+            "text/plain": [
+              "\"Ada Lovelace's initials are A. L.\""
+            ]
+          },
+          "metadata": {},
+          "output_type": "display_data"
+        }
+      ],
+      "source": [
+        "'Ada Lovelace\\'s initials are {}. {}.'.format('Ada'[0], 'Lovelace'[0])"
```
Comment (danielrazavi) [inline 01_slides/05_strings.ipynb:806]: This is a new syntax learners haven't been introduced to before. 

Previously it was:
```
first_name = "Ada"
print(first name[0])
```

Now we're showcasing it is possible to do string indexing directly:

```
first_letter = "Ada"[0]
```

I'm not against it, just wondering if this is a jump we should consider.

File: 01_slides/05_strings.ipynb [hunk 3]
```diff
@@ -609,6 +613,256 @@
...
+            "text/plain": [
+              "\"Ada Lovelace's initials are A. L.\""
+            ]
+          },
+          "metadata": {},
+          "output_type": "display_data"
+        }
+      ],
+      "source": [
+        "f'Ada Lovelace\\'s initials are {\"Ada\"[0]}. {\"Lovelace\"[0]}.'"
```
Comment (danielrazavi) [inline 01_slides/05_strings.ipynb:834]: Should we explain what `f` is? Should we link a "bonus reading"?

File: 01_slides/09_reading_and_writing_files.ipynb [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+   "outputs": [],
+   "source": [
+    "# check if a file exists to avoid overwriting it\n",
+    "text = 'Lavender is a small purple flower.'\n",
+    "\n",
+    "if os.path.exists('plants.txt'):\n",
+    "    print('plants.txt already exists')\n",
+    "else:\n",
+    "    with open('plants.txt', 'w') as f:\n",
+    "        f.write(text)"
```
Comment (danielrazavi) [inline 01_slides/09_reading_and_writing_files.ipynb:211]: We should add `plants.txt` into the `.gitignore`

**Review by danielrazavi — CHANGES_REQUESTED** [review]
Overall I don't have any issues with the idea, only minor questions and concerns here and there.

**Review by RohanAlexander — APPROVED** [review]

File: 01_slides/05_strings.ipynb (hunk 1 above)
Comment (kaylielau) [reply 01_slides/05_strings.ipynb:671]: As discussed on Slack, the string method content will be kept in this lesson to (1) provide additional insights into string manipulation and (2) allow students to use the methods introduced to complete Assignment 1.

File: 01_slides/05_strings.ipynb (hunk 3 above)
Comment (kaylielau) [reply 01_slides/05_strings.ipynb:834]: This has been modified in the slides.

- **f-string**: `f'...'` is an f-string, it allows for more concise and readable string interpolation by directly embedding expressions within the string.
- **`.format()` method**: This method allows for string formatting by specifying placeholders `{}` within the string and providing values to replace those placeholders in the `.format()` method.

Both expressions essentially achieve the same outcome: they insert the first characters of the `first_name` and `last_name` variables into the string to state Ada Lovelace's initials. The choice between them often comes down to personal preference and the Python version being used. Many developers prefer f-strings for their simplicity and readability.

File: 01_slides/05_strings.ipynb [hunk 5]
```diff
@@ -19,7 +19,10 @@
         "3. Multiline strings\n",
         "4. Escape sequences\n",
         "5. Working with strings and overloading\n",
-        "6. Strings within strings"
+        "6. Strings within strings\n",
```
[the remark this answers is not in the document]
Comment (kaylielau) [reply 01_slides/05_strings.ipynb:22]: This has been fixed.

**Comment by kaylielau** [comment]
@danielrazavi all your comments have been addressed.

**Review by danielrazavi — APPROVED** [review]

**Pull request merged**